src/keyword.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+/// Just like Clojure's Keyword. We'll need this as part of EDN parsing,
+/// but it's also used for identification within Datomish, so we'll define
+/// it here first.
+/// Callers are expected to follow these rules:
+/// http://www.clojure.org/reference/reader#_symbols
+#[derive(Clone,Debug,PartialEq)]
+pub struct Keyword {
+    pub name: String,
+    pub namespace: Option<String>,
+}
+
+impl Keyword {
+    pub fn new(name: &str) -> Self {
+        return Keyword { name: name.to_string(), namespace: None };
+    }
+
+    pub fn namespaced(name: &str, namespace: &str) -> Self {
+        return Keyword { name: name.to_string(), namespace: Some(namespace.to_string()) };
```

Comment:
Might as well check that `namespace` isn't empty, and that neither of these bits have `/` in them (really, are valid identifiers), and that we don't accept `::` (do we?).